pkg/profile/artifact.go

```diff
@@ -212,7 +219,40 @@ func (p *Profile) makeGitRepository() (*sourcev1.GitRepository, error) {
 	return gitRepo, nil
 }
 
+func (p *Profile) makeHelmRepository(url string, name string) (*sourcev1.HelmRepository, error) {
+	helmRepo := &sourcev1.HelmRepository{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.makeHelmRepoName(name),
+			Namespace: p.subscription.ObjectMeta.Namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       sourcev1.HelmRepositoryKind,
+			APIVersion: sourcev1.GroupVersion.String(),
+		},
+		Spec: sourcev1.HelmRepositorySpec{
+			URL: url,
+		},
+	}
+	err := controllerutil.SetControllerReference(&p.subscription, helmRepo, p.client.Scheme())
```

Comment:
now that we are wrapping all the resources in the `runtime.Object` can we do this in 1 place for all objects rather than repeating in each `makeResX`?

Reply:
Sure, yeah, we can do that. :)